sirepo/package_data/static/js/sirepo.js

```diff
@@ -1556,14 +1544,154 @@ SIREPO.app.service('validationService', function(utilities) {
 
 });
 
+SIREPO.app.factory('srCache', function(appState, $rootScope) {
+
+    // Browser side caching implemented using indexedDB
+    // - Caches sim frame responses
+    // - Allows cache to be cleared by simId or (simId, modelName)
+    // - Keeps MRU date on item access and deletes expired records at startup
+
+    const self = {};
+    const STORE = 'db';
+    const FRAME = 'frame';
+    // 30 days until expired
+    const EXPIRY_TIME = 30 * 24 * 60 * 60 * 1000;
+    let db = null;
+
+    function deleteKeys(keys) {
+        if (keys.length) {
+            const o = objectStore('readwrite');
+            if (o) {
+                for (const k of keys) {
+                    o.delete(k);
+                }
+            }
+        }
+    }
+
+    function invokeCallback(callback, value) {
+        $rootScope.$applyAsync(() => callback(value));
+    }
+
+    function key(prefix, value) {
+        return prefix + ':' + value;
+    }
+
+    function objectStore(mode) {
+        try {
+            if (db) {
+                return db.transaction(STORE, mode).objectStore(STORE);
+            }
+        }
+        catch (e) {
+        }
+        return null;
+    }
 
-SIREPO.app.factory('frameCache', function(appState, panelState, requestSender, authState, $interval, $rootScope, $timeout) {
+    function open() {
```

Comment:
Reading this is hard. I would prefer _open or _init. I know we don't have a style for this, but "open" looks like a primitive.

Reply:
I changed it from the generic open() to initializeDatabase() which is more descriptive.